### ISSUE TITLE
Output of individual losses (classification and localization) during training

### ISSUE THREAD

**Issue by rdaems** [pierluigiferrari/ssd_keras#196]
Hi @pierluigiferrari 

First of all thanks for this wonderful repo! Very well written and nicely documented.

I'm working on adding a third loss to the classification and localization loss: I want to train continuous values in the network, with a mean square loss. Every bounding box has then one or more continuous values attached, next to the classification.

In your implementation the classification and localization loss are combined in keras_ssd_loss.py, with a weighing factor alpha. I've added my third loss function there now and it seems to work. But I don't see a straightforward way of showing the individual losses during training to properly finetune the weighing between the three losses, and get more information during training.

In Keras it is possible to define multiple outputs, and combine multiple losses using the Keras API (https://keras.io/getting-started/functional-api-guide/#multi-input-and-multi-output-models). That way the individual loss values are exposed and you can see them during training.

I was wondering if you considered implementing the loss using a multi-output model? One big downside I see there is that I currently don't see a way of avoiding calculating positives and negatives three times then. I don't see a way to share that information over the three loss functions.

Would there be an other way of knowing the three losses individually during training?

**Comment by pierluigiferrari** [comment]
Thanks for the kinds words, I'm glad you find this implementation useful.

You don't need a multi-output model to solve your problem. Since you don't actually care about separate outputs but only about measuring the individual loss components to the overall loss, you can use the device that is intended for this purpose: [TensorFlow variable summaries](https://www.tensorflow.org/guide/summaries_and_tensorboard). They allow you to log the states our outputs of arbitrary TensorFlow operations and visualize them in TensorBoard.
